File: API/AdminFeatures/Players/Endpoints/CreatePlayer.cs
using API.DataAccess.Repositories;
using API.Domain.Models;

namespace API.AdminFeatures.Players.Endpoints;
public static class CreatePlayer
{
    public readonly record struct Request(string Name);
    public readonly record struct Response(int Id, string Name);
    public static async Task<IResult> HandleAsync(PlayerRepository repository, Request request)
    {
        // TODO: validation
        Player player = new() { Name = request.Name };
        Player result = await repository.CreateAsync(player);
        return Results.Ok(new Response(result.Id, result.Name));
    }
}

File: Web/src/lib/client/Players/CreatePlayer.ts
import { createEndpoint } from "../api";
export type CreatePlayerRequest = {
    readonly gameId: string;
    readonly name: string;
};
export type CreatePlayerResponse = {
    readonly id: number;
    readonly name: string;
};
export const CreatePlayer = createEndpoint<CreatePlayerRequest, CreatePlayerResponse>('/api/games/{gameId}/players', 'POST');
